src/main.rs

```diff
@@ -0,0 +1,148 @@
+use crate::postgres::Postgres;
+use actyxos_sdk::event_service::{EventService, Subscription};
+use anyhow::Result;
+use backtrace::Backtrace;
+use db::{Db, DbConnection};
+use env_logger::Env;
+use futures::{future::FutureExt, stream::StreamExt};
+use futures_batch::ChunksTimeoutStreamExt;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use structopt::StructOpt;
+use tokio_compat_02::FutureExt as FutureExt02;
+use tracing::*;
+
+mod db;
+mod postgres;
+
+#[derive(StructOpt, Debug, Serialize, Deserialize)]
+#[structopt(
+    about = "Insert events into a Postgres database.\nEnvironment variables:\n  AX_EVENT_SERVICE_URI: URL to connect to the Event Service. Default: http://localhost:4454/api/"
+)]
+struct Opt {
+    #[structopt(long, short = "f", takes_value = false, env = "FROM_START")]
+    from_start: bool,
+    #[structopt(long, short = "r", env, default_value = "1024")]
+    max_batch_records: usize,
+    #[structopt(long, short = "s", env, default_value = "1")]
+    max_batch_seconds: u64,
+    #[structopt(long, short = "d", env)]
+    db_name: String,
+    #[structopt(long, short = "h", env, default_value = "localhost")]
+    db_host: String,
+    #[structopt(long, short = "p", env, default_value = "5432")]
+    db_port: u16,
+    #[structopt(long, short = "u", env)]
+    db_user: String,
+    #[structopt(long, short = "w", env, hide_env_values = true)]
+    db_password: String,
+    #[structopt(long, short = "t", env, default_value = "events")]
+    table: String,
+    #[structopt(
+        about = "Subscriptions to subscribe to",
+        env,
+        parse(try_from_str = parse_subscriptions),
+        default_value = "{}"
+    )]
+    subscriptions: Vec<Subscription>,
+}
+
+fn parse_subscriptions(subs: &str) -> Result<Subscription, serde_json::Error> {
+    serde_json::from_str(subs)
+}
+
+#[tokio::main(flavor = "multi_thread", worker_threads = 2)]
+pub async fn main() -> Result<()> {
+    env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
+    init_panic_hook();
+    let opt: Opt = Opt::from_args();
+
+    info!("Subscribing to: {:?}", opt.subscriptions);
+
+    let pg = Postgres::new(
+        opt.db_host,
+        opt.db_port,
+        opt.db_user,
+        opt.db_password,
+        opt.db_name,
+        opt.table,
+    );
+
+    run_pipeline(
+        Box::new(pg),
+        opt.subscriptions,
+        opt.max_batch_records,
+        opt.max_batch_seconds,
+    )
+    .compat()
```

Comment:
why is this needed?

Reply:
Tokio 0.3 (I'll see if I can move it to 1.0 without much hassle). Before this it was: [`2be0853` (#1)](https://github.com/Actyx/events-to-db/pull/1/commits/2be085349ff7af27cac2b16172eb21152694ad58#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR74)